apps/service_providers/llm_service/runnables.py

```diff
@@ -197,6 +197,9 @@ def _chat_is_cancelled(self):
     def source_material(self):
         return self.experiment.source_material.material if self.experiment.source_material else ""
 
+    def _input_chain(self) -> Runnable[dict[str, Any], Any]:
```

Comment:
```suggestion
    def _input_chain(self) -> Runnable[dict[str, Any], Any]:
        """Return a langchain runnable that when invoked, will return
        the fully populated LLM input. This will be used during history compression.
        """
```